util/buildkitskipper/hasher/hasher.go

```diff
@@ -38,6 +45,10 @@ func (h *Hasher) HashCommand(cmd spec.Command) {
 	h.HashBytes(dt)
 }
 
+func (h *Hasher) HashInt(i int) {
+	h.h.Write([]byte(fmt.Sprintf("%d", i)))
```

Comment:
I think it's worthwhile that we include the datatype that we're hashing.

This will be a backwards-breaking change, but I think it's better to break things now rather than latter.

```suggestion
	h.h.Write([]byte(fmt.Sprintf("int:%d", i)))
```

and can you make a change to the HashString to include a `str:` prefix?

Reply:
Updated.

Reply:
awesome, thanks!